aptos-move/block-executor/src/view.rs

```diff
@@ -741,6 +732,70 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
         Ok((patched_bytes, mapping.into_inner()))
     }
 
+    fn extract_identifiers_from_value(
+        &self,
+        bytes: &Bytes,
+        layout: &MoveTypeLayout,
+    ) -> anyhow::Result<HashSet<T::Identifier>> {
+        let mapping = TemporaryExtractIdentifiersMapping::<T>::new();
+        let _patched_value =
+            deserialize_and_replace_values_with_ids(bytes.as_ref(), layout, &mapping).ok_or_else(
+                || anyhow::anyhow!("Failed to deserialize resource during id replacement"),
+            )?;
+        Ok(mapping.into_inner())
+    }
+
+    fn get_reads_needing_exchange_parallel(
+        &self,
+        read_set: &CapturedReads<T>,
+        delayed_write_set_keys: &HashSet<T::Identifier>,
+        skip: &HashSet<T::Key>,
+    ) -> BTreeMap<T::Key, (T::Value, Arc<MoveTypeLayout>)> {
+        read_set
+            .get_read_values_with_delayed_fields()
+            .filter(|(key, _)| !skip.contains(key))
+            .flat_map(|(key, data_read)| {
+                if let DataRead::Versioned(_version, value, Some(layout)) = data_read {
+                    if let Some(bytes) = value.bytes() {
+                        let identifiers_in_read = self
+                            .extract_identifiers_from_value(bytes, layout.as_ref())
+                            .unwrap();
```

Comment:
This fails if we have a problem with exchanges, logging an error and returning None like we do for other exchanges seems more uniform?